### PR TITLE
Add support for Ruby 2.3(.7)

### PR DESCRIPTION
**Pull request by Morriar**
This PR contains the few tweaks needed to support Ruby 2.3.x.

Please disregard the first commit as it's coming from #19.

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Why? Ruby 2.3 is not supported anymore even by the Ruby team.

**Comment by rafaelfranca** [comment]
Oh. We still need to support system Ruby on MacOS.

File: spec/tapioca/compilers/symbol_table_compiler_spec.rb [hunk 1]
```diff
@@ -1380,7 +1398,7 @@ def self.new(*args, &block)
                   super
                 end
 
-                instance_methods.each { |m| undef_method m unless /^__|^object_id$/.match?(m) }
+                instance_methods.each { |m| undef_method m unless /^__|^object_id$/.match(m) != nil }
```
Comment (rafaelfranca) [inline spec/tapioca/compilers/symbol_table_compiler_spec.rb]: don't need `!= nil` here. `unless /^__|^object_id$/.match(m)` will work.

Comment (Morriar) [reply spec/tapioca/compilers/symbol_table_compiler_spec.rb]: 🤦‍♂ fixed

**Comment by Morriar** [comment]
🎉

**Pull request merged**